R/object_name_linters.R

```diff
@@ -172,6 +206,25 @@ regexes_rd <- toString(paste0("\\sQuote{", names(style_regexes), "}"))
 #'   see the detailed note in [object_name_linter()] for more details.
 #'
 #' @param length maximum variable name length allowed.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "very_very_long_variable_name <- 1L",
```

Comment:
does `object_name_linters` work on `assign()` variables or `setGeneric()`?

If so, an example would be good, otherwise, maybe a follow-up issue?

Reply:
You mean this should lint?

``` r
library(lintr)

lint(
  text = assign("myVar", 2L),
  linters = object_name_linter(styles = c("dotted.case", "lowercase"))
)
```

<sup>Created on 2022-10-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
i think so, shouldn't it?

Reply:
Yes, I agree. Created https://github.com/r-lib/lintr/issues/1665 to track this.